docs/proposals/data-plane-config.md

```diff
@@ -1,21 +1,128 @@
 # Enhancement Proposal-929: Data Plane Dynamic Configuration
 
 - Issue: https://github.com/nginxinc/nginx-kubernetes-gateway/issues/929
-- Status: Provisional
+- Status: Implementable
 
 ## Summary
 
-This proposal is intended to contain the design for how to dynamically configure the data plane for the
-NGINX Gateway Fabric (NGF) project. Similar to control plane configuration, we should be able to leverage
+This proposal is intended to contain the design for how to dynamically configure global settings for the data plane
+of the NGINX Gateway Fabric (NGF) product. Similar to control plane configuration, we should be able to leverage
 a custom resource definition to define data plane configuration, considering fields such as telemetry and
 upstream zone size.
 
 ## Goals
 
-Define a CRD to dynamically configure various settings for the NGF data plane. The initial configurable options
-will be for telemetry (tracing) and upstream zone size.
+Define a CRD to dynamically configure various global settings for the NGF data plane. The initial configurable
+options will be for telemetry (tracing) and upstream zone size.
 
 ## Non-Goals
 
  1. This proposal is not defining every setting that needs to be present in the configuration.
  2. This proposal is not for any configuration related to control plane.
+
+## Introduction
+
+The NGF data plane will evolve to have various user-configurable options. These could include, but are not
+limited to, tracing, logging, or metrics. For the best user experience, these options should be able to be
+changed at runtime, to avoid having to restart NGF. The first set of options that we will allow users to
+configure are tracing and upstream zone size. The easiest and most intuitive way to implement a Kubernetes-native
+API is through a CRD.
+
+The purpose of this CRD is to contain "global" configuration options for the data plane, and not focused on policy
+per route or backend.
+
+In this doc, the term "user" will refer to the cluster operator (the person who installs and manages NGF). The
+cluster operator owns this CRD resource.
+
+## API, Customer Driven Interfaces, and User Experience
+
+The API would be provided in a CRD. An authorized user would interact with this CRD using `kubectl` to `get`
+or `edit` the configuration.
+
+Proposed configuration CRD example:
+
+```yaml
+apiVersion: gateway.nginx.org/v1alpha1
+kind: NginxProxy
+metadata:
+    name: nginx-proxy-config
+    namespace: nginx-gateway
+spec:
+    telemetry:
+        exporters:
+            otlp:
```

Comment:
This is to mimic the structure of [the otel collector config](https://opentelemetry.io/docs/collector/configuration/#exporters). It keeps us future-proof for if other protocols are supported.

Reply:
would it make sense to convert it to an array? Like an array of filters in HTTPRoute? https://github.com/nginxinc/nginx-gateway-fabric/blob/f547884c6596f8d915e09ff6c9af8a4b1253b63d/examples/http-header-filter/echo-route.yaml#L16 

Is it possible to define multiple destinations for otel in NGINX?

Reply:
The way the OTEL config is defined, you can have one of each exporter. By making `exporters` an object instead of an array, it allows you to define the proper sub-object like `otlp` or `prometheus` or `kafka`. I have yet to see support for multiple exporters of the same type sending to different backends.

Reply:
Right now the nginx otel module only supports `otlp`, and it may stay that way. So including that field may not be necessary for us.

 I could also do something like:
```
telemetry:
  tracing:
    endpoint....
```

Reply:
I was thinking about future proofing, but as I think about it more, I probably don't need the struct to be as complex and mimic the OTEL config. We are always going to be an exporter, and we'll likely just have the top level `tracing`, `metrics`, and `logging` options, which we can then configure protocol or type underneath that, instead of a top level `otlp` or `prometheus` object.